gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/query_service/specgen/ApiSpecGenQueryServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.query_service.specgen;
+
+import io.gravitee.apim.core.specgen.model.ApiSpecGen;
+import io.gravitee.apim.core.specgen.query_service.ApiSpecGenQueryService;
+import io.gravitee.definition.model.v4.ApiType;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.ApiRepository;
+import io.gravitee.repository.management.api.ScoringRulesetRepository;
+import io.gravitee.repository.management.model.Api;
+import io.gravitee.rest.api.service.ApiService;
+import io.gravitee.rest.api.service.common.ExecutionContext;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Rémi SULTAN (remi.sultan at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+
+@Service
+@Slf4j
+public class ApiSpecGenQueryServiceImpl implements ApiSpecGenQueryService {
```

Comment:
I must admit that I wouldn't have created the ApiSpecGenQueryServiceImpl but directly used the existing ApiCrudService.FindById and filtered in the UseCase
Optional Feedback